r/src/array_to_vector.cpp

```diff
@@ -656,8 +656,15 @@ class Converter_Null : public Converter {
     return Status::OK();
   }
 };
+
 std::shared_ptr<Converter> Converter::Make(const ArrayVector& arrays) {
```

Comment:
This should accept the type of `arrays` explicitly; a Converter should not fail when arrays is empty. Instead the result of conversion should also be empty.

Reply:
I updated https://issues.apache.org/jira/browse/ARROW-7798 to include this. I tried to quickly refactor what you suggest but it bubbles to rewriting almost all methods/functions of the file..

Reply:
It also lead me to find a bug with handling Dictionary ChunkedArray.